### ISSUE TITLE
color problem

### ISSUE THREAD

**Issue by Antonskoltech** [PRBonn/semantic-kitti-api#71]
Hello, I tested my own dataset on salasanext architecture, where I got labels and have used your tool to make a visualization. Actually, salsanext using VLP-64, and I did visualization for kitti labels which gave me good results, but no clue why it does not work on my dataset (point clouds from VLP32C), if this colores problem, or I got incorrect labels from salsanext, or I need to make changes in your config? thanks
![image](https://user-images.githubusercontent.com/44566122/110914500-839d0100-8327-11eb-95c9-ac9cf256253a.png)

**Comment by DavideCoppola97** [comment]
> Hello, I tested my own dataset on salasanext architecture, where I got labels and have used your tool to make a visualization. Actually, salsanext using VLP-64, and I did visualization for kitti labels which gave me good results, but no clue why it does not work on my dataset (point clouds from VLP32C), if this colores problem, or I got incorrect labels from salsanext, or I need to make changes in your config? thanks
> ![image](https://user-images.githubusercontent.com/44566122/110914500-839d0100-8327-11eb-95c9-ac9cf256253a.png)

Hi, how did you solve this problem? I have the same problem, using my custom dataset, although on a structural level the pointclouds are very similar to those of semanitcKitty, the result is similar to yours, with a predominance of red / orange also for the road, which should be purple and easily recognizable by all networks. I don't understand if it's a problem with my dataset or I'm doing something wrong in displaying the labels correctly.

**Comment by jbehley** [comment]
It's quite common that a method trained on one sensor does not perform well on another sensor. As SalsaNext use a projection the "content" of the projection is totally different (i.e. the vertical field-of-view, the number of beams, etc,) While image-based methods are fairly robust to changes in resolution (the images are anyway downsampled), point cloud-based methods struggle with a different sensor setup.

For debugging, I would look also at the histogram (counts per label) and verify that a specific label is mainly appearing.

**Comment by DavideCoppola97** [comment]
> It's quite common that a method trained on one sensor does not perform well on another sensor. As SalsaNext use a projection the "content" of the projection is totally different (i.e. the vertical field-of-view, the number of beams, etc,) While image-based methods are fairly robust to changes in resolution (the images are anyway downsampled), point cloud-based methods struggle with a different sensor setup.
> 
> For debugging, I would look also at the histogram (counts per label) and verify that a specific label is mainly appearing.
Thanks so much for the quick reply!
The strange thing is that I got my data through the lidar of a simulator, so I could set exactly the same number of beams, fovs, etc. of the velodyne64 used to build the SemanticKitty dataset.
Having not yet been able to fine tune, I expected problems in recognizing objects, which in the simulator (Carla) may be different from those in real life, but I did not expect the network to fail to recognize even the road, which should be the thing. easier to recognize, yet the result of the segmentation is almost entirely wrong. I have had similar results with various networks, Rangenet, SqueezeSegV3 (in which from the first tests almost all the points are marked with the label corresponding to the red color, or at least that's what I see from the visualize) etc ...
I am attaching a sample photo
![Cattura](https://user-images.githubusercontent.com/63714542/130275424-6c21a224-f1f9-430b-927a-57429b927540.JPG)
